### ISSUE TITLE
Wrong npm tag on pre-release

### ISSUE THREAD

**Issue by MoonLiightz** [release-it/release-it#586]
Hi guys, I'm not sure if this is a "bug" or if I have misunderstood something. See the following scenario:

The version in my package.json: `"version": "0.2.1-rc.0"`. 
Now I want to release a new rc-version (0.2.1-rc.1) and execute `release-it --preRelease=rc --npm.tag=next`, but I get this warning `WARNING No version found in npm registry. Assuming new package.`. Just for your information, the version detection (0.2.1-rc.1) works anyway and the package is in the registry, it's just the warning. I did a little debugging (with `--dry-run`) and found out, that release-it looks for the wrong npm tag: `$ npm show <hidden-name>@rc version` (rc instead of next).
After some more debugging, I found that when I changed the following function
https://github.com/release-it/release-it/blob/47548b52f6d2c5b89f1b14dcfb16e946d4ca536c/lib/plugin/npm/npm.js#L164-L167
into
```js
  async resolveTag(version) {
    const { isPreRelease, preReleaseId } = parseVersion(version);
    return !isPreRelease ? DEFAULT_TAG : (await this.guessPreReleaseTag());
  }
```
it works as expected and the warning is no longer shown.
A `--dry-run` shows now `$ npm show <hidden-name>@next version` 

I didn't go deeper into the code, so I can't judge if it's a good idea to adopt this change or what it might do. Maybe I just misunderstood something and that's why I get the warning. But if not, I can create a PR if you wish.

**Comment by webpro** [comment]
Looks like you understand everything perfectly. Definitely sounds like a bug and I'll investigate/fix shortly. Thanks!

**Comment by webpro** [comment]
Should be fixed in [v12.5.0](https://github.com/release-it/release-it/releases/tag/12.5.0)

**Comment by MoonLiightz** [comment]
Can I confirm, thanks for your work.
